src/plugins/guided_onboarding/public/plugin.tsx

```diff
@@ -41,12 +44,13 @@ export class GuidedOnboardingPlugin
     { cloud }: AppPluginStartDependencies
   ): GuidedOnboardingPluginStart {
     const { chrome, http, theme, application, notifications } = core;
+    const { ui: isGuidedOnboardingUiEnabled } = this.ctx.config.get<ClientConfigType>();
 
     // Initialize services
     apiService.setup(http, !!cloud?.isCloudEnabled);
```

Comment:
I think we can add the config boolean to the setup and prevent any requests to be sent to the API if guided onboarding is disabled. Requests might be sent on the pages that check for guided onboarding state for their specific guide step, for example Rules in Security.
```suggestion
    apiService.setup(http, !!cloud?.isCloudEnabled && isGuidedOnboardingUiEnabled);
```

Reply:
👍 good idea